src/App.js

```diff
@@ -32,8 +34,21 @@ class App extends React.Component {
       this.history.push(path)
     }
   }
+  isAppInstalled(appId) {
+    return (
+      appId === 'home' ||
+      appId === 'settings' ||
+      !!apps.find(app => app.id === appId)
+    )
+  }
+  handleNavigateBack = () => {
+    this.state.lastPath ? this.history.goBack() : this.history.replace('/')
+  }
   handleNavigation = location => {
-    this.setState({ path: location.pathname })
+    this.setState({
+      lastPath: this.state.path,
```

Comment:
@bpierre not the most elegant, but the easiest way I could think of detecting whether someone was in a "fresh" start of the app.

AFAIK there's no native way to examine browser history due to security issues.